src/Illuminate/Database/Eloquent/HasCollection.php

```diff
@@ -15,6 +18,27 @@ trait HasCollection
      */
     public function newCollection(array $models = [])
     {
-        return new static::$collectionClass($models);
+        $collectionClass = $this->resolveCollectionAttribute() ?? static::$collectionClass;
+
+        return new $collectionClass($models);
+    }
+
+    /**
+     * Resolve the collection class name from the CollectedBy attribute.
+     *
+     * @return ?class-string<TCollection>
```

Comment:
The short null syntax is not valid in PHPDocs.
```suggestion
     * @return class-string<TCollection>|null
```